mne/annotations.py

```diff
@@ -214,6 +214,37 @@ def __iadd__(self, other):
                                                  other.orig_time))
         return self.append(other.onset, other.duration, other.description)
 
+    def __iter__(self):
+        for idx in range(len(self.onset)):
+            yield (self.onset[idx], self.duration[idx], self.description[idx])
```

Comment:
I thought we said we wanted to use dict now?

Reply:
I understand and agree with using dict when slicing with a single element. 

> slice gets you Annotations and int gets you a dict

Mainly because if we return a tuple then we break backward compatibility if we change it later.

But I missed that you were also proposing to change the iterator. The entire purpose of this PR was to be able to rewrite 

https://github.com/mne-tools/mne-python/blob/68410a1468b7fa5d72ec631a844a1b448caf8223/mne/annotations.py#L798-L802

into 

```py
for onset, duration, description in annotations[event_sel]:
    # do stuff
```

but you cannot do `for onset, duration, description in annotations[event_sel].values()`

But I might be doing something wrong

```py
    class Ob_tuple(object):
        def __init__(self):
            self.NUM = 5
            self.a = np.array([x for x in range(self.NUM)])
            self.b = np.array([x.__repr__() for x in range(self.NUM)])

        def __iter__(self):
            for ii in range(self.NUM):
                yield (self.a[ii], self.b[ii])

    for a, b in Ob_tuple():
        print((a, b))

    class Ob_dict(object):
        def __init__(self):
            self.NUM = 5
            self.a = np.array([x for x in range(self.NUM)])
            self.b = np.array([x.__repr__() for x in range(self.NUM)])

        def __iter__(self):
            for ii in range(self.NUM):
                yield {'a': self.a[ii], 'b': self.b[ii]}

    # the dict elements are ok
    for x in Ob_dict():
        print(x)

    # iterate over a dict gives you the keys not the values
    for a, b in Ob_dict():
        print((a, b))
```